x-pack/plugins/security_solution/server/lib/detection_engine/rule_execution_log/event_log_adapter/event_log_client.ts

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IEventLogger, IEventLogService } from '../../../../../../event_log/server';
+import { RuleExecutionStatus } from '../../../../../common/detection_engine/schemas/common/schemas';
+import { LogStatusChangeArgs } from '../types';
+import { RuleExecutionLogAction, RULE_EXECUTION_LOG_PROVIDER } from './constants';
+
+const statusSeverityDict: Record<RuleExecutionStatus, number> = {
+  [RuleExecutionStatus.succeeded]: 0,
+  [RuleExecutionStatus['going to run']]: 10,
+  [RuleExecutionStatus.warning]: 20,
+  [RuleExecutionStatus['partial failure']]: 20,
+  [RuleExecutionStatus.failed]: 30,
+};
+
+interface FindExecutionLogArgs {
+  ruleIds: string[];
+  spaceId: string;
+  logsCount?: number;
+  statuses?: RuleExecutionStatus[];
+}
+
+interface LogExecutionMetricsArgs {
+  ruleId: string;
+  spaceId: string;
+  metrics: EventLogExecutionMetrics;
+}
+
+interface EventLogExecutionMetrics {
+  searchDurationSum?: number;
+  indexingDurationSum?: number;
+  executionGapDuration?: number;
+}
+
+interface IEventLogClient {
+  find: (args: FindExecutionLogArgs) => Promise<{}>;
+  logStatusChange: (args: LogStatusChangeArgs) => Promise<void>;
+  logExecutionMetrics: (args: LogExecutionMetricsArgs) => Promise<void>;
+}
+
+export class EventLogClient implements IEventLogClient {
```

Comment:
Why do we need to separate `EventLogClient` and `EventLogAdapter`?

Reply:
`EventLogAdapter` is needed for backward compatibility with legacy data structures. So long as we have to keep two alternative implementations (SO-based and eventLog-based), we should keep "glue" code somewhere that transforms Event Log data structures to the legacy format. 

As soon as we are confident that eventLog-based implementation is fully operational and tested, we can throw away the SO-based implementation, all adapters, and other glue-code. After that point, we can introduce new API endpoints along with breaking changes to the data format. Most likely, that will happen only after 7.16 FF.